include/nil/crypto3/zk/commitments/detail/polynomial/basic_fri.hpp

```diff
@@ -281,7 +285,7 @@ namespace nil {
                           const std::size_t fri_step) {
 
                     if (f.size() != D->size()) {
-                        f.resize(D->size());
+                        f.resize(D->size(), nullptr, D);
```

Comment:
Here we don't want nullptr, these polynomials are very frequently of the same size, we better create caches for them as well.